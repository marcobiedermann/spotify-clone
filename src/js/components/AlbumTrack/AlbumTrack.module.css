.track__name {
  color: #fff;
}
