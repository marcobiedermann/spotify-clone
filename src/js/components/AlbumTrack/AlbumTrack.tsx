import format from 'date-fns/format';
import React from 'react';
import { Link } from 'react-router-dom';
import styles from './AlbumTrack.module.css';

interface Artist {
  name?: string;
  id?: string;
}

export interface AlbumTrackProps {
  artists?: Artist[];
  duration_ms?: number;
  /* eslint-disable-next-line react/no-unused-prop-types */
  id?: string;
  name?: string;
  track_number?: number;
}

function AlbumTrack(props: AlbumTrackProps): JSX.Element {
  const { artists = [], duration_ms = 0, name, track_number } = props;

  return (
    <tr>
      <td>{track_number}</td>
      <td>
        <div className={styles.track__name}>{name}</div>
        <div>
          {artists
            .map<React.ReactNode>((artist) => (
              <Link key={artist.id} to={`/artists/${artist.id}`}>
                {artist.name}
              </Link>
            ))
            .reduce((prev, curr) => [prev, ', ', curr])}
        </div>
      </td>
      <td align="right">{format(duration_ms, 'm:ss')}</td>
    </tr>
  );
}

export default AlbumTrack;
