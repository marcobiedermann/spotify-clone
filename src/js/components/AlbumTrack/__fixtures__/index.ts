const albumTrack = {
  artists: [
    {
      id: '5VQE4WOzPu9h3HnGLuBoA6',
      name: 'Vlasta Marek',
    },
  ],
  duration_ms: 730066,
  id: '64IL0ZrOV7F6BDPm5fhfUa',
  name: 'Is',
  track_number: 2,
};

export default albumTrack;
