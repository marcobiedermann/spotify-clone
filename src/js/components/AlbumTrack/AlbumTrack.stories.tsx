import type { Meta, StoryObj } from '@storybook/react';
import React from 'react';
import { MemoryRouter } from 'react-router-dom';
import AlbumTrack from '.';
import albumTrackFixtures from './__fixtures__';

const meta: Meta<typeof AlbumTrack> = {
  component: AlbumTrack,
  args: albumTrackFixtures,
  decorators: [
    (Story) => (
      <MemoryRouter>
        <table>
          <tbody>
            <Story />
          </tbody>
        </table>
      </MemoryRouter>
    ),
  ],
};

type Story = StoryObj<typeof AlbumTrack>;

const Default: Story = {};

export { Default };
export default meta;
