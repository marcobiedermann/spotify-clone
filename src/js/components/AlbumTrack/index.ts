import AlbumTrack from './AlbumTrack';

export default AlbumTrack;
export * from './AlbumTrack';
