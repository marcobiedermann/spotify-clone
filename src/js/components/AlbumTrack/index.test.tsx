import { render } from '@testing-library/react';
import React from 'react';
import { MemoryRouter } from 'react-router-dom';
import AlbumTrack from '.';
import albumTrackFixtures from './__fixtures__';

it('renders correctly', () => {
  const { container } = render(
    <MemoryRouter>
      <AlbumTrack {...albumTrackFixtures} />
    </MemoryRouter>,
  );

  expect(container.firstChild).toMatchSnapshot();
});
