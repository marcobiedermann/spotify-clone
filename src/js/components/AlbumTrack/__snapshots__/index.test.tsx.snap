// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly 1`] = `
<tr>
  <td>
    2
  </td>
  <td>
    <div
      class="track__name"
    >
      Is
    </div>
    <div>
      <a
        href="/artists/5VQE4WOzPu9h3HnGLuBoA6"
      >
        Vlasta Marek
      </a>
    </div>
  </td>
  <td
    align="right"
  >
    12:10
  </td>
</tr>
`;
